### ISSUE TITLE
ICU 67.1 brings in new C fallback method in the API

### ISSUE THREAD

**Issue by filmil** [google/rust_icu#59]
Between 66.0.1 and 67.1 there was a change in how locale fallback works in the ICU4C API, which fixed a long-standing bug
with the library: https://unicode-org.atlassian.net/browse/ICU-20931

Now that the bug is fixed, our language matching tests no longer work, we should fix that.

```
---- tests::test_accept_language_exact_match stdout ----
thread 'tests::test_accept_language_exact_match' panicked at 'assertion failed: `(left == right)`
  left: `(Some(ULoc { repr: "es_MX" }), ULOC_ACCEPT_FALLBACK)`,
 right: `(Some(ULoc { repr: "ar_EG" }), ULOC_ACCEPT_VALID)`', rust_icu_uloc/src/lib.rs:774:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by filmil** [comment]
@kpozin something for you?

**Comment by filmil** [comment]
Note that 67.1 is not yet released. Perhaps we should just modify our instructions to check out ICU not-at-master?

**Comment by filmil** [comment]
Looks like 67.1 is released.

**Comment by filmil** [comment]
FYI this is now fixed with #76
